common/src/watcher.rs

```diff
@@ -84,3 +85,20 @@ where
     });
     rx
 }
+
+// Replacement for pipe_async function in eventuals
+// Listen to the changes in a receiver and runs parametric function
+pub fn watch_pipe<T, F, Fut>(rx: watch::Receiver<T>, function: F) -> JoinHandle<()>
+where
+    T: Clone + Send + Sync + 'static,
+    F: Fn(watch::Receiver<T>) -> Fut + Send + Sync + 'static,
```

Comment:
The Function should be `Fn(&T) -> Fut` or (if needed) `Fn(T) -> Fut` whichever has less changes in the codebase.

Reply:
Hi, I thought of it, I think there's a small tradeoff here, it'd look cool but this way we could skip expensive cloning in many places? what do you think?

Reply:
if cloning is not needed in all places, you could just use the reference for it and input the `watcher.borrow()`.

Reply:
unfortunately cloning is required in some places, so I'll update the I'll change it to `Fn(T) -> Fut` itself.

Reply:
But if I'm not mistaken, you can clone the reference inside the function

Reply:
yes, but cloning seems essential in some places but not in others, reference would do fine in some. I'll try to make all work using `Ref` itself, if it's not possible I'll use cloning itself, that'll be good right?

Reply:
sure can [✌️]